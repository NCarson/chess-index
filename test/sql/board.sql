
select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1'::board::text
      ='rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1';

select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq -'::board;
select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq -'::board::text
      ='rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq -';

select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w - -'::board::text
      ='rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w - -';

select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR b - e3'::board::text
      ='rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR b - e3';

select '8/8/8/8/8/8/8/8'::board::text;

--XXX probably should not allow this
select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KKKKKQ -'::board::text;

-- should fail
select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - -1 1'::board::text;
select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 500 1'::board::text;
select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 1 550'::board::text;
select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 1 -1'::board::text;
select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR . KQkq -'::board::text;
select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w . -'::board::text;

