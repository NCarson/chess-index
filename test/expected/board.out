select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1'::board::text
      ='rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1';
 ?column? 
----------
 t
(1 row)

select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq -'::board;
                        board                         
------------------------------------------------------
 rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq -
(1 row)

select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq -'::board::text
      ='rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq -';
 ?column? 
----------
 t
(1 row)

select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w - -'::board::text
      ='rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w - -';
 ?column? 
----------
 t
(1 row)

select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR b - e3'::board::text
      ='rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR b - e3';
 ?column? 
----------
 t
(1 row)

select '8/8/8/8/8/8/8/8'::board::text;
         text          
-----------------------
 8/8/8/8/8/8/8/8 w - -
(1 row)

--XXX probably should not allow this
select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KKKKKQ -'::board::text;
                        text                        
----------------------------------------------------
 rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQ -
(1 row)

-- should fail
select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - -1 1'::board::text;
ERROR:  bad move number in fen footer 'w KQkq - -1 1' at char '-'
LINE 1: select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq -...
               ^
select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 500 1'::board::text;
ERROR:  halfmove clock number '500' is greater than the maximum of 127:
LINE 1: select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq -...
               ^
select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 1 550'::board::text;
ERROR:  move number '550' is greater than maximum of 511:
LINE 1: select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq -...
               ^
select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 1 -1'::board::text;
ERROR:  bad halfmove clock number in fen footer 'w KQkq - 1 -1' at char '-'
LINE 1: select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq -...
               ^
select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR . KQkq -'::board::text;
ERROR:  bad move side in fen: '.'
LINE 1: select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR . KQkq -...
               ^
select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w . -'::board::text;
ERROR:  bad castle availability in fen: '.'
LINE 1: select 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w . -'::...
               ^
